mypy/types.py

```diff
@@ -2920,16 +2920,23 @@ def visit_type_var(self, typ: TypeVarType) -> Type:
         return typ
 
 
+def replace_alias_location(tp: Type, newline: int, newcolumn: int) -> Type:
+    """Set line and column attributes in a generic type and arguments recursively.
+    """
+    tp.line = newline
+    tp.column = newcolumn
+    for ta in getattr(tp, 'args', []):
+        replace_alias_location(ta, newline, newcolumn)
```

Comment:
This might not find all component types (e.g. within callables). Also, we prefer to avoid `getattr` since it can't be type checked by mypy and mypyc generates slow code for it. 

Can you instead define a subclass of `TypeTraverserVisitor`, and override at least `visit_instance` to update line/column info. You can use it to update all component type objects (of the desired types), and in a type-safe manner.

Reply:
@JukkaL I've updated the PR, please could you check again?